src/app-gocardless/banks/norwegian-se-norwnok1.js

```diff
@@ -0,0 +1,93 @@
+import {
+  printIban,
+  amountToInteger,
+  sortByBookingDateOrValueDate,
+} from '../utils.js';
+
+/** @type {import('./bank.interface.js').IBank} */
+export default {
+  institutionId: 'NORWEGIAN_SE_NORWNOK1',
+
+  normalizeAccount(account) {
+    return {
+      account_id: account.id,
+      institution: account.institution,
+      mask: account.iban.slice(-4),
+      iban: account.iban,
+      name: [account.name, printIban(account)].join(' '),
+      official_name: account.product,
+      type: 'checking',
+    };
+  },
+
+  normalizeTransaction(transaction, booked) {
+    /**
+     * The way Bank Norwegian handles the date fields is rather strange and
+     * countrary to GoCardless's documentation.
+     *
+     * For booked transactions Bank Norwegian sends a `valueDate` field that
+     * doesn't match the NextGenPSD2 definition of `valueDate` which is what we
+     * expect to receive from GoCardless.  Therefore we remove the incorrect
+     * field so that transactions are correctly imported.
+     */
+    if (booked) {
+      delete transaction.valueDate;
+      return transaction;
+    }
+
+    /**
+     * For pending transactions there are two possibilities.  Either the
+     * transaction has a `valueDate`, in which case the `valueDate` we receive
+     * corresponds to when the transaction actually occurred, and so we simply
+     * return the transaction as-is.
+     */
+    if (transaction.valueDate) {
+      return transaction;
+    }
+
+    /**
+     * If the pending transaction didn't have a `valueDate` field then it
+     * should have a `remittanceInformationStructured` field which contains the
+     * date we expect to receive as the `valueDate`.  In this case we extract
+     * the date from that field and set it as `valueDate`.
+     */
+    if (transaction.remittanceInformationStructured) {
+      const remittanceInfoRegex = /(?:[^ ]+ ){5}(\d{4}-\d{2}-\d{2})/;
```

Comment:
Sorry for blurring the screenshot, but the real `remittanceInformationStructured` looks like this with the credit card number:
`111111******2222  SEK 39 SEK 39 2023-08-04 00:20:23 SL Stockholm SE    , 39, 2023-08-08T00:00:00.000`

so it seems we would be [safe enough to go with](https://regex101.com/r/CB7M6c/1)
```suggestion
      const remittanceInfoRegex = /(\d{4}-\d{2}-\d{2})/;
```